apps/managedidentity/managedidentity.go

```diff
@@ -260,7 +269,14 @@ func New(id ID, options ...ClientOption) (Client, error) {
 		httpClient:         opts.httpClient,
 		retryPolicyEnabled: opts.retryPolicyEnabled,
 		source:             source,
+		logger:             opts.logger,
+		piiLogging:         opts.piiLogging,
 	}
+
+	logOnce.Do(func() {
+		client.logger.Log(context.Background(), slog.LevelInfo, "Managed Identity", slog.String("source", string(client.source)))
```

Comment:
logging may entail network I/O, so this could still block indefinitely depending on the handler implementation. It would be better to do this in `AcquireToken`